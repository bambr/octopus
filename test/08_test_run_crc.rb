#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
run_crc_delay = 0.4
EOD
  end
end


Env.connect_eval do |env|
  ping
  sleep 0.5 # run_crc_delay + 0.1 tolerance
  33.times do |i|
    insert [i, i + 1, i + 2]
  end

  wait_for { open('|./octopus --cat 00000000000000000002.xlog 2>/dev/null').each_line.grep(/run_crc/).length > 0 }
  env.stop

  puts `./octopus --cat 00000000000000000002.xlog 2>/dev/null | sed 's/tm:[^ ]* //' | grep run_crc` + "\n"
end
