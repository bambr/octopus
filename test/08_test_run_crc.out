# box.ping()
:pong

# box.insert([0, 1, 2])
1

# box.insert([1, 2, 3])
1

# box.insert([2, 3, 4])
1

# box.insert([3, 4, 5])
1

# box.insert([4, 5, 6])
1

# box.insert([5, 6, 7])
1

# box.insert([6, 7, 8])
1

# box.insert([7, 8, 9])
1

# box.insert([8, 9, 10])
1

# box.insert([9, 10, 11])
1

# box.insert([10, 11, 12])
1

# box.insert([11, 12, 13])
1

# box.insert([12, 13, 14])
1

# box.insert([13, 14, 15])
1

# box.insert([14, 15, 16])
1

# box.insert([15, 16, 17])
1

# box.insert([16, 17, 18])
1

# box.insert([17, 18, 19])
1

# box.insert([18, 19, 20])
1

# box.insert([19, 20, 21])
1

# box.insert([20, 21, 22])
1

# box.insert([21, 22, 23])
1

# box.insert([22, 23, 24])
1

# box.insert([23, 24, 25])
1

# box.insert([24, 25, 26])
1

# box.insert([25, 26, 27])
1

# box.insert([26, 27, 28])
1

# box.insert([27, 28, 29])
1

# box.insert([28, 29, 30])
1

# box.insert([29, 30, 31])
1

# box.insert([30, 31, 32])
1

# box.insert([31, 32, 33])
1

# box.insert([32, 33, 34])
1

lsn:34 scn:34 t:sys/run_crc 0.0.0.0:0 SCN:32 log:0xc0240090

